lib/puppet/provider/tomcat_install/linux.rb

```diff
@@ -0,0 +1,98 @@
+require 'fileutils'
+require 'rubygems/package'
+require 'zlib'
+require 'open-uri'
+
+Puppet::Type.type(:tomcat_install).provide(:linux) do
+  def initialize(value={})
+    super(value)
+  end
+
+  def get_tomcat_version(catalina_home)
+    return nil unless File.exist?("#{catalina_home}/RELEASE-NOTES")
+
+    version = nil
+
+    File.open("#{catalina_home}/RELEASE-NOTES", 'r').each_line do |li|
+      if li[/Version \d{1,2}\.\d{1,2}\.\d{1,2}/]
+        version = li[/\d{1,2}\.\d{1,2}\.\d{1,2}/]
+        break
+      end
+    end
+
+    version
+  end
+
+  def exists?
+    get_tomcat_version(resource[:catalina_home]) == resource[:version]
+  end
+
+  def destroy
+    FileUtils.rm_r resource[:catalina_home] if Dir.exist?(resource[:catalina_home])
+  end
+
+  def create
+    destroy
```

Comment:
It looks like `create` will be called if `version` doesn't match the installed version. If create calls `destroy` here, that would wipe out the entire `catalina_home`... but what if `catalina_base` of a `tomcat::instance` is the same as the home? This would wipe out all instance content as well. Is there a possible workaround for this scenario?